### PR TITLE
Various initial improvements

### PR DESCRIPTION
**Pull request by arwer13**
- state config format
  - disallow not checksummed addresses
  - allow big numbers as numbers
- refactor
  - simplify code
  - better error messages in deep asserts
- features
  - download ABIs automatically
  - generate config boilerplate from seed config
  - OpenZeppelin non enumerable ACL checks

### CODE REVIEW

File: src/state-mate.ts [hunk 1]
```diff
@@ -1,15 +1,22 @@
-import { existsSync, readFileSync } from "node:fs";
+// import { existsSync, readFileSync, writeFileSync, copyFileSync } from "node:fs";
+import fs from "node:fs";
 import path from "node:path";
+import https from 'https';
+import http from 'http';
```
Comment (tamtamchik) [inline src/state-mate.ts:5]: No need, since `fetch` is global. See comments below.
```suggestion
```

File: src/state-mate.ts [hunk 2]
```diff
@@ -421,22 +560,230 @@ async function checkNetworkSection(section: NetworkSection, sectionTitle: string
...
+  const implementationAddress = contractInfo["Implementation"];
+  if (implementationAddress) {
+    implementation = await _loadContractInfoFromEtherscanExplorer(implementationAddress, explorerHostname, explorerKey);
+  }
+
+  return { contractName: contractInfo["ContractName"], abi, address, implementation };
+}
+
+async function loadContractInfoFromExplorer(address: string, explorerHostname: string, explorerKey?: string): Promise<ContractInfoFromExplorer> {
+  if (explorerHostname.indexOf("mode.network") > -1) {
```
Comment (tamtamchik) [inline src/state-mate.ts:635]: Why are they differently threated? It looks like both use the Etherscan-based. Do we need a separate check for BSC?

**Review by tamtamchik — COMMENTED** [review]
For some reason `--save-abi-from-explorer` does not work for me. 

```yarn start ./configs/binance/sepolia-testnet.yaml -o l1 --save-abi-from-explorer
ERROR: TypeError: Cannot read properties of undefined (reading 'value')
Trace
    at logError (/Users/.../state-mate/src/state-mate.ts:247:11)
    at /Users/.../state-mate/src/state-mate.ts:822:3
```

Worth moving to `fetch` for HTTP/HTTPS requests. The change will simplify the code and make working with the network really simple.

I plan to revisit the PR when I will sort out the error with downloading ABI.

File: src/state-mate.ts (hunk 2 above)
Comment (arwer13) [reply src/state-mate.ts:629]: Unfortunately, part of them are slightly different, the same problem is illustrated in diffyscan: https://github.com/lidofinance/diffyscan/blob/525cc6319515791a1f32906b33f710471ff34506/diffyscan/utils/explorer.py#L80-L87

Comment (tamtamchik) [reply src/state-mate.ts:629]: Wow... didn't expect that. Thanks, will know now!

**Review by tamtamchik — APPROVED** [review]
Added some tweaks that jumped on me while testing PR locally in https://github.com/lidofinance/state-mate/pull/11
Feel free to merge or close #11 if not needed. 

This PR is approved.

**Pull request merged**